crates/storage/src/schema/revision_0057.rs

```diff
@@ -0,0 +1,1637 @@
+                        let receipt = zstd::decode_all(receipt.as_slice())
+                            .context("Decompressing receipt")
+                            .unwrap();
+                        let (receipt, _): (dto::Receipt, _) = bincode::serde::decode_from_slice(
+                            &receipt,
+                            bincode::config::standard(),
+                        )
+                        .context("Deserializing receipt")
+                        .unwrap();
+                        dto::TransactionWithReceipt {
+                            transaction,
+                            receipt,
+                        }
+                    })
+                    .collect();
+                let events: Vec<_> = events
+                    .into_iter()
+                    .map(|events| {
+                        let events = zstd::decode_all(events.as_slice())
+                            .context("Decompressing events")
+                            .unwrap();
+                        let (events, _): (dto::Events, _) =
+                            bincode::serde::decode_from_slice(&events, bincode::config::standard())
+                                .context("Deserializing events")
+                                .unwrap();
+                        events
+                    })
+                    .collect();
+                let transactions =
+                    bincode::serde::encode_to_vec(transactions, bincode::config::standard())
+                        .context("Serializing transaction")
+                        .unwrap();
+                let transactions = compressor
+                    .compress(&transactions)
+                    .context("Compressing transaction")
+                    .unwrap();
+                let events = bincode::serde::encode_to_vec(events, bincode::config::standard())
+                    .context("Serializing events")
+                    .unwrap();
+                let events = compressor
+                    .compress(&events)
+                    .context("Compressing events")
+                    .unwrap();
+
+                // Store the updated values.
+                if let Err(err) = insert_tx.send((block_number, transactions, events, len)) {
+                    panic!("Failed to send transaction: {:?}", err);
+                }
+            }
+        });
+        transformers.push(transformer);
+    }
+
+    let count = tx.query_row("SELECT COUNT(*) FROM starknet_transactions", [], |row| {
+        row.get::<_, i64>(0)
+    })?;
+    tx.execute_batch(
+        r"
+        CREATE TABLE transactions (
+            block_number INTEGER NOT NULL,
+            transactions BLOB NOT NULL,
+            events BLOB
+        );
+        CREATE INDEX transactions_block_number ON transactions(block_number);
+        CREATE TABLE transaction_hashes (
+            hash         BLOB PRIMARY KEY NOT NULL,
+            block_number INTEGER NOT NULL
```

Comment:
Right now we don't remove rows from this table when purging block data. We could do that automatically by making block_number a foreign key and using `ON DELETE CASCADE`.

```suggestion
            block_number INTEGER NOT NULL REFERENCES block_headers(number) ON DELETE CASCADE
```